lib/jobs/generate-tag.js

```diff
@@ -41,8 +41,13 @@ function generateTagJobFactory(
     function GenerateTagJob(options, context, taskId) {
         GenerateTagJob.super_.call(this, logger, options, context, taskId);
 
-        this.nodeId = context.target || options.nodeId;
-        assert.isMongoId(this.nodeId, 'context.target || options.nodeId');
+        if (options.nodes) {
+            this.nodes = options.nodes;
+            assert.arrayOfString(this.nodes, 'options.nodes');
+        } else {
+            this.nodeId = context.target || options.nodeId;
```

Comment:
Actually the logic can be combined, so that the `_run` only handles the multiple nodes case. Even in future, we want to get rid of the singular nodeId, it will be easier:
```javascript
if (options.nodes) {
    this.nodes = options.nodes;
} else {
    this.nodes = [context.target || options.nodeId];
} 
assert.arrayOfString(this.nodes, 'Nodes from options');
``